repos/keg-components/src/hooks/useOutsideDetect.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from 'react'
+import { getPlatform } from 'KegGetPlatform'
+
+const platform = getPlatform()
+
+/**
+ * 
+ * @param {Event} evt - event object
+ * @param {RefObject} ref - element to check 
+ * @returns {boolean} - true if event occured outside of element stored at ref.current
+ */
+const eventIsOutside = (evt, ref) => {
+  return ref?.current && !ref.current.contains(evt.target)
+}
+
+/**
+ * Listens for the user clicking outside the element stored in `ref`. If it detects a click,
+ * then it calls onOutsideClick.
+ * @param {RefObject} ref - ref object storing a DOM Node
+ * @param {Function?} onOutsideClick - fn of form (event) => {}
+ */
+export const useOutsideDetect = (ref, onOutsideClick) => {
+  useEffect(() => {
+    if (platform !== 'web') 
```

Comment:
See my comments above for the `useKeyPress` hook

Reply:
updated